src/IncrementalChecker.ts

```diff
@@ -439,19 +434,18 @@ export class IncrementalChecker implements IncrementalCheckerInterface {
     // get all lints
     const lints = this.files
       .keys()
-      .reduce(
+      .reduce<tslint.RuleFailure[]>(
         (innerLints, filePath) =>
           innerLints.concat(this.files.getData(filePath).tslints),
-        [] as RuleFailure[]
+        []
       );
 
-    // normalize and deduplicate lints
-    return NormalizedMessage.deduplicate(
-      lints.map(this.createNormalizedMessageFromRuleFailure)
-    );
+    return createIssuesFromTsLintRuleFailures(lints);
   }
 
-  public getEsLints(cancellationToken: CancellationToken) {
+  public async getEsLintIssues(
+    cancellationToken: CancellationToken
+  ): Promise<Issue[]> {
```

Comment:
I missed this first time round, what's the motivation behind making the method `async`? It doesn't look like it consumes any `async` methods? (Though I am reviewing on a phone so might have missed it)

Reply:
I wanted to make the interface consistent - we have async method for `getTypeScriptIssues` so we should have the same approach to other integrations. Thanks to that we are not leaking internal implementation to the public interface - we assume the "worst" scenario that every integration is potentially asynchronous.